### PR TITLE
Extract server and framework using new prefix

### PR DESCRIPTION
**Pull request by echanlee**
Use new prefixes to extract information. Have not tested with master because downloader has not been updated.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -112,10 +112,20 @@ function downloadDeps(callback: Function) {
...
             let l = lines[i];
-            if (l.startsWith('Got ')) {
-                let p = l.split(' ')[1];
-                if (!server) server = p;
-                else framework = p;
+            const serverPrefix = "Got language server: ";
+            const frameworkPrefix = "Got Checker Framework: ";
+            if (l.startsWith(serverPrefix)) {
+                const p = l.substring(serverPrefix.length);
+                if (!server) {
```
Comment (wmdietl) [inline src/extension.ts:119]: Now that this uses the full prefix, do we still need this check?
Should the framework branch below have similar checks?

**Review by wmdietl — COMMENTED** [review]
In separate PRs, can you please 1) add automatic formatting of the typescript sources in this repository and 2) figure out how to better integrate the CF-downloader.
Thanks!

File: src/extension.ts (hunk 1 above)
Comment (echanlee) [reply src/extension.ts:119]: I think this isn't necessary. I will remove it.

**Comment by wmdietl** [comment]
@echanlee Can you please pull in master so that this contains the changes from #86 ? There are some conflicts.

File: src/extension.ts [hunk 2]
```diff
@@ -128,10 +128,12 @@ function downloadDeps(callback: Function) {
...
       let l = lines[i];
-      if (l.startsWith("Got ")) {
-        let p = l.split(" ")[1];
-        if (!server) server = p;
-        else framework = p;
+      const serverPrefix = "Got language server: ";
+      const frameworkPrefix = "Got Checker Framework: ";
+      if (l.startsWith(serverPrefix)) {
+        server = l.substring(serverPrefix.length);
+      } else if (l.substring(frameworkPrefix.length)) {
```
Comment (wmdietl) [inline src/extension.ts:135]:
```suggestion
      } else if (l.startsWith(frameworkPrefix)) {
```

File: src/extension.ts [hunk 3]
```diff
@@ -128,10 +128,12 @@ function downloadDeps(callback: Function) {
...
+        framework = l.substring(frameworkPrefix.length);
       }
```
Comment (wmdietl) [inline src/extension.ts:137]:
```suggestion
      } else {
        // TODO raise an error for unexpected responses
      }
```

**Review by wmdietl — COMMENTED** [review]
@echanlee Thanks for the update!
Does my first suggested update look good?
Can you expand the second change and actually raise an error?

**Comment by wmdietl** [comment]
> @echanlee Can you please pull in master so that this contains the changes from #86 ? There are some conflicts.

@echanlee Could you pull out the changes to README.md into a separate PR? Those seem unrelated to this PR and are likely left-over from #86. Thanks!

File: src/extension.ts (hunk 3 above)
Comment (wmdietl) [reply src/extension.ts:137]: Actually, other output possible. So don't issue an error, but add a comment.

File: src/extension.ts [hunk 4]
```diff
@@ -128,10 +128,14 @@ function downloadDeps(callback: Function) {
     let lines = str.split(/(\r?\n)/g);
     for (let i = 0; i < lines.length; ++i) {
       let l = lines[i];
-      if (l.startsWith("Got ")) {
-        let p = l.split(" ")[1];
-        if (!server) server = p;
-        else framework = p;
+      const serverPrefix = "Got language server: ";
```
Comment (wmdietl) [inline src/extension.ts:131]:
```suggestion
      // See https://github.com/eisopux/checker-framework-languageserver-downloader/blob/master/src/main/java/org/checkerframework/languageserver/DownloaderMain.java#L50
      const serverPrefix = "Got language server: ";
```

File: src/extension.ts [hunk 5]
```diff
@@ -128,10 +128,14 @@ function downloadDeps(callback: Function) {
...
+      const frameworkPrefix = "Got Checker Framework: ";
+      if (l.startsWith(serverPrefix)) {
+        server = l.substring(serverPrefix.length);
+      } else if (l.startsWith(frameworkPrefix)) {
+        framework = l.substring(frameworkPrefix.length);
+      } else {
+        // TODO raise an error for unexpected responses
```
Comment (wmdietl) [inline src/extension.ts:138]:
```suggestion
        // TODO handle other progress/error responses
```

**Pull request merged**